### PR TITLE
Add article about creating a cluster

### PR DESCRIPTION
**Pull request by kmsholcomb**
I created an article with the steps for creating a cluster in Carina. It borrows from Everett's getting started article and Carolyn's credentials doc. Please tech review the heck out of this!

### CODE REVIEW

**Comment by kmsholcomb** [comment]
Should the intro mention that clusters run Docker Swarm?

**Comment by adrianotto** [comment]
That detail is not important in this context. I am concerned that the article makes reference to a specific version of the docket client. As we change the version of swarm and docker on the server side, it will obsolete is text. What is our plan for keeping it correct over time?

--
Adrian

On Oct 18, 2015, at 9:57 AM, Kelly Holcomb <notifications@github.com<mailto:notifications@github.com>> wrote:


Should the intro mention that clusters run Docker Swarm?

-

**Comment by rgbkrk** [comment]
> As we change the version of swarm and docker on the server side, it will obsolete is text. What is our plan for keeping it correct over time?

We'll also be at the behest of the latest and greatest version of the docker client, which may be more recent than what's installed on our swarm clusters. Because of this perpetual delay, we're going to have to keep updating the document over time. Acting like the version isn't a problem is only going to cause more problems than simply telling them to get a specific version.

@smashwilson and I put together a [docker version manager](https://github.com/rgbkrk/dvm) to handle the version drift directly. Handling the case of "what happens when you have a week old swarm cluster or docker machine, your client is updated, and you want to access that docker host?" We can't necessarily force the ideology of dvm on people, so we have to continue to document the latest version.

File: _tutorials/create-connect-cluster.md [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+
+1. Save the zip file to a location on your computer.
+
+    The name of the zip file is the same as the name of your cluster.
+
+1. Unzip the file.
+
+    The name of the directory that is created is the same as the name of the cluster.
+
+1. If you are using Windows and use CMD or PowerShell, follow the instructions in the [Create a Windows script](#windows) section and then precede to the next step.
```
Comment (stanzikratel) [inline _tutorials/create-connect-cluster.md]: Does the next step mean Step 5?

File: _tutorials/create-connect-cluster.md [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+
+This tutorial shows you how to create a cluster in Carina and then download its credentials so that you can securely connect to the cluster. 
+
+A cluster is a pool of compute, storage, and networking resources that serves as a host for one or more containerized applications. Clusters are secured with Transport Layer Security (TLS) certificates. Each cluster has its own set of credentials, which you can download in a zip file from the Carina control panel. The credentials zip file contains the following files:
+
+* ca.pem - Certificate Authority, used by clients to validate servers
+* cert.pem - Client Certificate, used by clients to identify themselves to servers
+* key.pem - Client Private Key, used by clients to encrypt their requests
+* ca-key.pem - Certificate Authority Key, private file used to generate more client certificates
+* docker.env - Shell environment configuration script
```
Comment (carolynvs) [inline _tutorials/create-connect-cluster.md]: We now have docker.ps1 and docker.cmd files as well! :-) So we should tweak this and the rcs creds article to include them.

File: _tutorials/create-connect-cluster.md [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+### Prerequisite
+
+A Carina account. If you do not already have one, follow the [sign up process](https://mycluster.rackspacecloud.com/managed).
+
+### Create a cluster
+
+1. Log in to [http://mycluster.rackspacecloud.com](http://mycluster.rackspacecloud.com).
+1. In the **Create New** field, enter a name for the cluster.
+1. Click **Create Cluster**.
+1. Click the **Refresh** button until your cluster reaches a status of **active**.
```
Comment (carolynvs) [inline _tutorials/create-connect-cluster.md]: This documents our current UI but if Keith's work goes out in time, we'll need to circle back and make sure it still matches.

File: _tutorials/create-connect-cluster.md [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+
+1. Open a terminal application.
+
+1. Change to the credentials directory. For example:
+
+    ```bash
+    $ cd /Users/octopus/Downloads/mycluster
+    ```
+
+1. Download the Docker 1.8.2 client for your OS into the credentials directory.
```
Comment (carolynvs) [inline _tutorials/create-connect-cluster.md]: It's like chasing a runaway train but we are using 1.8.3 now. :smile:

File: _tutorials/create-connect-cluster.md [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+      └ Reserved Memory: 0 B / 4.2 GiB
+      └ Labels: executiondriver=native-0.2, kernelversion=3.18.21-1-rackos, operatingsystem=Debian GNU/Linux 7 (wheezy) (containerized), storagedriver=aufs
+    CPUs: 12
+    Total Memory: 4.2 GiB
+    Name: 3e867f7a955f
+    ```
+
+    **Note:** A newly created cluster contains three containers that are necessary for cluster management. For more information about these containers, see [Introduction to Docker Swarm](/docs/tutorials/005-docker-swarm-intro).  
+
+### <a name="windows"></a> Create a Windows script
```
Comment (carolynvs) [inline _tutorials/create-connect-cluster.md]: We should remove this section. I've worked with Thomas and Ash to get these built into the credentials zip and that's now on the production site.

File: _tutorials/create-connect-cluster.md [hunk 6]
```diff
@@ -0,0 +1,154 @@
...
+1. On Linux and Mac OS X, ensure that the client is executable.
+
+    ```bash
+    $ chmod u+x docker
+    ```
+
+1. Load the cluster credentials and configuration.
+    * (_Linux and Mac OSX users_) Run `source docker.env`.
+    * (_Windows CMD users_) Run `docker.cmd`.
+    * (_Windows PowerShell users_) Run `docker.ps1`.
```
Comment (carolynvs) [inline _tutorials/create-connect-cluster.md]: This command may fail for most users the first time. I have a workaround that fixes the security necessary to run powershell scripts in load-docker-environment-on-windows.md which we should include in our Troubleshooting section.

**Comment by carolynvs** [comment]
@kmsholcomb :+1: Looks good, added a few comments/gotchas. Later we'll need to go back and update this and the other articles (e.g. rcs-creds) when we are ready to use DVM and the Carina CLI.

File: _tutorials/create-connect-cluster.md (hunk 4 above)
Comment (rgbkrk) [inline _tutorials/create-connect-cluster.md]: Choo choo!

**Comment by everett-toews** [comment]
I've made a lot of updates to the [Getting started on Carina](https://github.com/rackerlabs/docs-container-service/blob/master/_tutorials/getting-started-on-carina.md) tutorial to have it work with Windows. I'm going to be doing some more testing on it tomorrow. Once it's solid we can reuse it here.

**Comment by kmsholcomb** [comment]
Sounds good. I've been holding off on making updates until you finalized the zero to hero. Let me know when I can leverage your updated content. :)

**Comment by kmsholcomb** [comment]
Hi @everett-toews: Is the Getting started article done? Can I start stealing from it?

**Comment by everett-toews** [comment]
I think the cluster interaction stuff is solid (section "Connect to your cluster" and down) but the whole thing can't be finalized until the sign up process is done and the web app has been deployed.

**Comment by kmsholcomb** [comment]
OK. Today I will get it as updated as I can and merge. That way it's in the GitHub file structure. Then it can be updated whenever we have that info. Thanks!

**Pull request merged**